vminitd/Sources/vminitd/Server+GRPC.swift

```diff
@@ -921,6 +921,72 @@ extension Initd: Com_Apple_Containerization_Sandbox_V3_SandboxContextAsyncProvid
             $0.result = r
         }
     }
+
+    func notifyFileSystemEvent(
+        request: Com_Apple_Containerization_Sandbox_V3_NotifyFileSystemEventRequest,
+        context: GRPCAsyncServerCallContext
+    ) async throws -> Com_Apple_Containerization_Sandbox_V3_NotifyFileSystemEventResponse {
+        log.debug(
+            "notifyFileSystemEvent",
+            metadata: [
+                "path": "\(request.path)",
+                "eventType": "\(request.eventType)",
+            ])
+
+        do {
+            try await generateSyntheticInotifyEvent(
+                path: request.path,
+                eventType: request.eventType
+            )
+
+            return .with {
+                $0.success = true
+            }
+        } catch {
+            log.error(
+                "notifyFileSystemEvent",
+                metadata: [
+                    "error": "\(error)"
+                ])
+
+            return .with {
+                $0.success = false
+                $0.error = error.localizedDescription
+            }
+        }
+    }
+
+    private func generateSyntheticInotifyEvent(
+        path: String,
+        eventType: Com_Apple_Containerization_Sandbox_V3_FileSystemEventType
+    ) async throws {
+        switch eventType {
+        case .modify:
+            // Touch file to update timestamp -> generates IN_ATTRIB event
+            let now = Date()
+            try FileManager.default.setAttributes(
```

Comment:
We shouldn't change the modification date. The modification date was already changed, we just want to notify.